app/helpers/form_helper.rb

```diff
@@ -40,31 +40,53 @@ def crud_form(object, *attrs, &block)
 
     buttons_bottom = options.delete(:buttons_bottom)
     buttons_top = options.delete(:buttons_top) { true }
-    submit_label = options.delete(:submit_label)
-    cancel_url = get_cancel_url(object, options)
+    form_button_options = options.slice(:add_another_label, :add_another, :submit_label)
+                                 .merge(cancel_url: get_cancel_url(object, options))
 
     standard_form(object, options) do |form|
       content = form.error_messages
 
-      content << save_form_buttons(form, submit_label, cancel_url, 'top') if buttons_top
+      content << form_buttons(form, form_button_options.merge(toolbar_class: 'top')) if buttons_top
 
       content << if block_given?
                    capture(form, &block)
                  else
                    form.labeled_input_fields(*attrs)
                  end
 
-      content << save_form_buttons(form, submit_label, cancel_url, 'bottom') if buttons_bottom
+      if buttons_bottom
+        content << form_buttons(form, form_button_options.merge(toolbar_class: 'bottom'))
+      end
 
       content.html_safe
     end
   end
 
-  def save_form_buttons(form, submit_label, cancel_url, toolbar_class = nil)
-    submit_label ||= ti(:"button.save")
-    content_tag(:div, class: "btn-toolbar #{toolbar_class}") do
-      submit_button(form, submit_label) +
-      cancel_link(cancel_url)
+  def button_toolbar(form, toolbar_class:, &block)
+    content_tag(:div, toolbar_class: "btn-toolbar #{toolbar_class}") do
+      capture(form, &block)
+    end
+  end
+
+  def form_buttons(form, submit_label: ti(:"button.save"), cancel_url: nil, toolbar_class: nil,
+                          add_another: false, add_another_label: ti(:"button.add_another"))
+    button_toolbar(form, toolbar_class: toolbar_class) do
+      content = submit_button(form, submit_label)
+      content << add_another_button(form, add_another_label) if add_another.present?
+      content << cancel_link(cancel_url) if cancel_url.present?
+
+      content.html_safe
```

Comment:
```suggestion
      content
```
Unnötiges `.html_safe` lieber weglassen, falls die Methode in Zukunft erweitert wird